lms/djangoapps/oauth_dispatch/dot_overrides.py

```diff
@@ -41,3 +42,24 @@ def _authenticate(self, username, password):
             else:
                 authenticated_user = authenticate(username=email_user.username, password=password)
         return authenticated_user
+
+    def save_bearer_token(self, token, request, *args, **kwargs):
+        """
+        Ensure that access tokens issued via client credentials grant are associated with the owner of the
+        ``Application``.
+        """
+        grant_type = request.grant_type
+        user = request.user
+
+        if grant_type == 'client_credentials':
+            # Temporarily remove the grant type to avoid triggering the super method's code that removes request.user.
```

Comment:
Thanks.  That helps.  Hmm.. but still don't know why we need to override this though.  The `client_credentials` grant is a server-to-server access, and doesn't need to be associated with a user, right?